doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,500 @@
+integer Points; these operations have been wrapped in `toIntRounded()`:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp toInt
+
+- Rectangles are constructed by two Points --- the top-left and the bottom-right
+corners.
+
+- By far the most important thing here is solving an issue that sometimes
+detector spits out coordinates located outside of the image; if we pass such an
+ROI to be processed, errors of the landmarks detector will occure. The frame box
+`borders` is created and then intersected with the face rect (by `operator&()`)
+to handle these cases and save the ROI in `outFaces` which is for sure
+inside the frame.
+
+## Facial landmarks detector inference and post-processing {#gapi_fb_landm_detect}
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp ld_inf
+
+First of all, a few words about custom types: the `Contour` type is used for
+arrays of points which are naturally contours (ordered sets of points):
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp cont
+
+The array with all landmarks is an array of points; syntactically it is a `Contour`
+though semantically is not a contour
+so I use another type `Landmarks` which syntactically is equal to `Contour`
+type to underline semantical differences:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp landm
+
+Post-processing of the Mat is required here too: there are 35 landmarks given by
+the detector for each face in a frame; the first 18 of them are facial elements
+(eyes, eyebrows, a nose, a mouth) and the last 17 --- a jaw contour. Landmarks
+are floating point values of coordinates normalized relatively to an input ROI
+(not the original frame). What is of the same importance, we need contours of
+eyes, mouths, faces, etc., not the landmarks, for the further goals.
+So the process is split into two parts.
+
+### Normalized points scaling {#gapi_fb_ld_scl}
+
+The first step is to denormalize landmarks' coordinates to the real pixel
+coordinates of the source frame. Also we are able to divide points into two
+groups here --- face elements and a jaw contour. This is a kernel that does the
+described actions:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp ld_pp_scl
+
+Some points:
+
+- It takes in a `cv::Mat` of detections and the array of the ROI's from the
+previous step; it gives out an array of facial elements' points' arrays and an
+array of jaw contours.
+
+- Data parsing is the same --- by pointer.
+
+- We iterate by 2 in one step because for every point `Point.x` and `Point.y`
+are contained successively.
+
+- Since the coordinates are related to an input ROI, the coordinates of the
+ROI's top-left corner should be added.
+
+### Contours getting {#gapi_fb_ld_cnts}
+
+The more definite idea of this face beautification algorithm is to smooth out the
```

Comment:
I suppose you can rephrase `The more definite idea` to make it sound better

Reply:
OK